components/script/dom/htmlinputelement.rs

```diff
@@ -434,21 +438,17 @@ impl HTMLInputElement {
         };
 
         match ty {
+            // Step 3.1: it's a button but it is not submitter.
             atom!("submit") | atom!("button") | atom!("reset") if !is_submitter => return None,
-            atom!("radio") | atom!("checkbox") => {
-                if !self.Checked() || name.is_empty() {
-                    return None;
-                }
-            },
+            // Step 3.1: it's the "Checkbox" or "Radio Button" and whose checkedness is false.
+            atom!("radio") | atom!("checkbox") if !self.Checked() || name.is_empty() => return None,
             atom!("image") | atom!("file") => return None, // Unimplemented
-            _ => {
-                if name.is_empty() {
-                    return None;
-                }
-            }
+            // Step 3.1: it's not the "Image Button" and doesn't have a name attribute.
+            _ if name.is_empty() => return None
```

Comment:
This part here is what's generating the compilation error. You removed the `_` match case and added a pattern guard here.

This has different semantics than the code you deleted. Previously it meant "match against all other values of `ty`, and if name is empty, return None". Right now it means "match against all other values of `ty` that has their name empty, and return None". See the difference?